objectivec/GPBDictionary.m

```diff
@@ -13579,23 +13579,27 @@ - (void)removeObjectForKey:(id)aKey {
 
 - (id)copyWithZone:(NSZone *)zone {
   if (_dictionary == nil) {
-    _dictionary = [[NSMutableDictionary alloc] init];
+    return [[NSMutableDictionary allocWithZone:zone] init];
   }
   return [_dictionary copyWithZone:zone];
 }
 
 - (id)mutableCopyWithZone:(NSZone *)zone {
   if (_dictionary == nil) {
-    _dictionary = [[NSMutableDictionary alloc] init];
+    return [[NSMutableDictionary allocWithZone:zone] init];
   }
   return [_dictionary mutableCopyWithZone:zone];
 }
 
 - (id)objectForKeyedSubscript:(id)key {
+  // Not really needed, but subscripting is likely common enough it doesn't hurt
+  // to ensure it goes directly to the real NSMutableDictionary.
   return [_dictionary objectForKeyedSubscript:key];
 }
 
 - (void)setObject:(id)obj forKeyedSubscript:(id<NSCopying>)key {
+  // Not really needed, but subscripting is likely common enough it doesn't hurt
+  // to ensure it goes directly to the real NSMutableDictionary.
```

Comment:
perhaps move this comment to the setObject:forKeyedSubscript: call? I got confused at first when I saw the nil check just below